python/demo/poisson/demo_poisson.py

```diff
@@ -134,10 +141,19 @@
 # immediately above. The definition of the Dirichlet boundary condition
 # then looks as follows: ::
 
+# Create a function that returns True if the point is on the boundary
+
+
+def marker(x):
+    left = np.isclose(x[0], 0.0)
+    right = np.isclose(x[0], 2.0)
+    return np.logical_or(left, right)
+
+
 # Define boundary condition on x = 0 or x = 1
-facets = locate_entities_boundary(mesh, 1, lambda x: np.logical_or(np.isclose(x[0], 0.0),
-                                                                   np.isclose(x[0], 2.0)))
-bc = dirichletbc(ScalarType(0), locate_dofs_topological(V, 1, facets), V)
+facets = _mesh.locate_entities_boundary(mesh, dim=1, marker=marker)
```

Comment:
These few steps will need some extra explanation in the comments.

Reply:
I added a few more explanations in https://github.com/FEniCS/dolfinx/pull/1979/commits/b48e86cbdefc812f07b0e2f131a7d799853168fe but these should probably be revised by someone who know the intentions behind the choice of API here. 

Personally I think it would be good not having to call `locate_entities_boundary` and `locate_dofs_topological`, but I guess this is also me not being used to the new syntax in FEniCSx.

Reply:
The reason for splitting entities and degrees of freedom is that the entities can be used for a lot of other things than applying Dirichlet boundary conditions.
For Dirichlet bcs there are two functions; locate_dofs_topological and locate_dofs_geometrical.
Locate_dofs_geometrical only make sense for spaces that has point evaluations as functionals (i.e. Lagrange), and cannot be used with nedelec spaces and others (as a degree of freedom is associated with a facet).
For such spaces you can use locate_dofs_topological.